src/main/java/net/minecraftforge/server/permission/context/ContextKey.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission.context;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+
+import javax.annotation.Nonnull;
+
+public final class ContextKey<T>
+{
+    public static final ContextKey<BlockPos> BLOCK_POS = new ContextKey<BlockPos>("blockpos");
+    public static final ContextKey<IBlockState> BLOCK_STATE = new ContextKey<IBlockState>("blockstate");
```

Comment:
All lowercase may cause confusions. Use camel case or underscore to make understanding easier.